src/clients/Deluge.ts

```diff
@@ -262,36 +317,46 @@ export default class Deluge implements TorrentClient {
 	 * returns information needed to complete/validate injection
 	 */
 	private async getTorrentInfo(searchee: Searchee): Promise<TorrentInfo> {
+		if (!searchee.infoHash) {
+			throw new Error("Can't search a torrent without a infoHash");
+		}
+
+		let torrent: TorrentInfo;
 		try {
-			let torrentLabel: string = undefined;
 			const params = [
 				["state", "progress", "save_path", "label"],
 				{ hash: searchee.infoHash },
 			];
-			const response = await this.call("web.update_ui", params);
-			if (response?.result?.torrents?.[searchee.infoHash] === undefined) {
+
+			const response = await this.call<TorrentStatus>(
+				"web.update_ui",
+				params
+			);
+
+			if (response && typeof response.result === "object") {
```

Comment:
```suggestion
			if (response.result) {
```

Reply:
I think this makes the else unreachable

Reply:
should we just delete the `else` then?

regarding my suggestion, when would `result` be truthy but not `typeof result === "object"`? I think that would only happen if result were a number.

Reply:
When the torrent is not present, you get the following 
```{
  result: {
    connected: true,
    torrents: {},
    filters: {
      state: [Array],
      tracker_host: [Array],
      owner: [Array],
      label: [Array]
    },```

Reply:
result would always seem to exist unless something catastrophic happened, perhaps we should throw and attempt to reconnect Deluge or at the very least reauth and validate?

Reply:
I think just assuming result exists is fine. Other errors probably wouldn't go through this codepath